go/beacon/api/scrape.go

```diff
@@ -0,0 +1,144 @@
+	Participants  uint64 `json:"participants"`
+	Threshold     uint64 `json:"threshold"`
+	PVSSThreshold uint64 `json:"pvss_threshold"`
+
+	CommitInterval  int64 `json:"commit_interval"`
+	RevealInterval  int64 `json:"reveal_interval"`
+	TransitionDelay int64 `json:"transition_delay"`
+
+	GasCosts transaction.Costs `json:"gas_costs,omitempty"`
+}
+
+const (
+	// GasOpSCRAPECommit is the gas operation identifier for a commit.
+	GasOpSCRAPECommit transaction.Op = "scrape_commit"
+	// GasOpSCRAPEReveal is the gas operation identifier for a reveal.
+	GasOpSCRAPEReveal transaction.Op = "scrape_reveal"
+)
+
+// DefaultGasCosts are the "default" gas costs for operations.
+var DefaultGasCosts = transaction.Costs{
+	GasOpSCRAPECommit: 1000,
+	GasOpSCRAPEReveal: 1000,
+}
+
+// SCRAPECommit is a SCRAPE commitment transaction payload.
+type SCRAPECommit struct {
+	Epoch EpochTime `json:"epoch"`
+	Round uint64    `json:"round"`
+
+	Commit *scrape.Commit `json:"commit,omitempty"`
+}
+
+// SCRAPEReveal is a SCRAPE reveal transaction payload.
+type SCRAPEReveal struct {
+	Epoch EpochTime `json:"epoch"`
+	Round uint64    `json:"round"`
+
+	Reveal *scrape.Reveal `json:"reveal,omitempty"`
+}
+
+// RoundState is a SCRAPE round state.
+type RoundState uint8
+
+const (
+	StateInvalid  RoundState = 0
+	StateCommit   RoundState = 1
+	StateReveal   RoundState = 2
+	StateComplete RoundState = 3
+)
+
+func (s RoundState) String() string {
+	switch s {
+	case StateInvalid:
+		return "invalid"
+	case StateCommit:
+		return "commit"
+	case StateReveal:
+		return "reveal"
+	case StateComplete:
+		return "complete"
+	default:
+		return fmt.Sprintf("[invalid state: %d]", s)
+	}
+}
+
+// SCRAPEState is the SCRAPE backend state.
+type SCRAPEState struct {
+	Height int64 `json:"height,omitempty"`
+
+	Epoch EpochTime  `json:"epoch,omitempty"`
+	Round uint64     `json:"round,omitempty"`
+	State RoundState `json:"state,omitempty"`
+
+	Instance     *scrape.Instance      `json:"instance,omitempty"`
+	Participants []signature.PublicKey `json:"participants,omitempty"`
+	Entropy      []byte                `json:"entropy,omitempty"`
+
+	BadParticipants map[signature.PublicKey]bool `json:"bad_participants,omitempty"`
+
+	CommitDeadline   int64 `json:"commit_deadline,omitempty"`
+	RevealDeadline   int64 `json:"reveal_deadline,omitempty"`
+	TransitionHeight int64 `json:"transition_height,omitempty"`
+
+	RuntimeDisableHeight int64 `json:"runtime_disable_height,omitempty"`
+}
+
+// SCRAPEEvent is a SCRAPE backend event.
+type SCRAPEEvent struct {
+	Height int64 `json:"height,omitempty"`
+
+	Epoch EpochTime  `json:"epoch,omitempty"`
+	Round uint64     `json:"round,omitempty"`
+	State RoundState `json:"state,omitempty"`
+
+	Participants []signature.PublicKey `json:"participants,omitempty"`
+}
+
+func (ev *SCRAPEEvent) FromState(state *SCRAPEState) {
+	ev.Height = state.Height
+	ev.Epoch = state.Epoch
+	ev.Round = state.Round
+	ev.State = state.State
+	ev.Participants = state.Participants
```

Comment:
Eh, "don't do that then".  The routine's only in the API because it's used by both the consensus backend and the node worker, which both are written correctly.

The beacon in general isn't intended to be a general use externally exposed component anyway...